src/master-playlist-controller.js

```diff
@@ -1134,16 +1143,30 @@ export class MasterPlaylistController extends videojs.EventTarget {
         media.resolvedUri;
       return this.mediaSource.endOfStream('decode');
     }
-    this.mainSegmentLoader_.mimeType(mimeTypes[0]);
-    if (mimeTypes[1]) {
-      this.audioSegmentLoader_.mimeType(mimeTypes[1]);
-    }
 
+    this.configureLoaderMimeTypes_(mimeTypes);
     // exclude any incompatible variant streams from future playlist
     // selection
     this.excludeIncompatibleVariants_(media);
   }
 
+  configureLoaderMimeTypes_(mimeTypes) {
+    // If the content is demuxed, we can't start appending segments to a source buffer
+    // until both source buffers are set up, or else the browser may not let us add the
+    // second source buffer (it will assume we are playing either audio only or video
+    // only).
+    const sourceBufferEmitter =
+      // if the first mime type has muxed video and audio then we shouldn't wait on the
+      // second source buffer
+      mimeTypes.length > 1 && mimeTypes[0].indexOf(',') === -1 ?
+        new videojs.EventTarget() : null;
```

Comment:
we might want to think about expanding this emitter concept to its own controller class. When we start tearing apart flash and pulling the muxer into vhs and removing contrib-media-sources, we will still need a place for some of the contrib-media-sources logic to live e.g. buffer management logic that was added to address the edge/ie corruption issues

Reply:
I agree that we may end up needing that, but for now I think it's best to hold off. We may not get to it for a while, and if we do get to it sooner, it may end up having different requirements than we suppose right now.